.phpcs.xml.dist

```diff
@@ -0,0 +1,71 @@
+	<!-- For VIP GO PHPCS severity levels: https://vip.wordpress.com/documentation/phpcs-review-feedback/ -->
+
+	<!-- What to scan -->
+	<file>./themes/</file>
+	<!-- Ignoring Files and Folders:
+		https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-files-and-folders -->
+	<exclude-pattern>./themes/twentytwentyone/</exclude-pattern>
+	<exclude-pattern>./node_modules/</exclude-pattern>
+	<exclude-pattern>./vendor/</exclude-pattern>
+
+	<!-- How to scan -->
+	<arg value="sp"/> <!-- Show sniff and progress -->
+	<arg name="colors"/> <!-- Show results with colors. Disable if working on Windows -->
+	<arg name="basepath" value="."/> <!-- Strip the file paths down to the relevant bit -->
+	<arg name="parallel" value="8"/> <!-- Enables parallel processing when available for faster results -->
+	<arg name="extensions" value="php,js"/> <!-- Limit to PHP and JS files -->
+
+	<!-- Rules: Check PHP version compatibility - see
+		https://github.com/PHPCompatibility/PHPCompatibilityWP -->
+	<rule ref="PHPCompatibilityWP"/>
+	<!-- For help in understanding this testVersion:
+		https://github.com/PHPCompatibility/PHPCompatibility#sniffing-your-code-for-compatibility-with-specific-php-versions -->
+	<config name="testVersion" value="7.4-"/>
+
+	<!-- Rules: VIP Coding Standards - see
+		https://github.com/Automattic/VIP-Coding-Standards -->
+	<rule ref="WordPress-VIP-Go"/>
+
+	<!-- Rules: WordPress Coding Standards - see
+		https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards -->
+	<!--<rule ref="WordPress-Extra"/>--> <!-- Includes WordPress-Core -->
+	<!--<rule ref="WordPress-Docs"/>-->
+	<!-- For help in understanding this minimum_supported_wp_version:
+		https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties#setting-minimum-supported-wp-version-for-all-sniffs-in-one-go-wpcs-0140 -->
+	<config name="minimum_supported_wp_version" value="5.9-"/>
+
+	<!-- For help in understanding this prefixes property:
+		https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties#naming-conventions-prefix-everything-in-the-global-namespace -->
+	<rule ref="WordPress.NamingConventions.PrefixAllGlobals">
+		<properties>
+			<property name="prefixes" type="array">
+				<element value="your_theme_prefix"/> <!-- Change this value to your theme slug/prefix. -->
+				<!--<element value="my_custom_plugin"/>-->
+			</property>
+		</properties>
+	</rule>
+
+	<!-- For help in understanding this text_domain property:
+		https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties#internationalization-setting-your-text-domain -->
+	<rule ref="WordPress.WP.I18n">
+		<properties>
+			<property name="text_domain" type="array">
+				<element value="your-theme-slug"/> <!-- Change this value to your theme slug. -->
+				<!--<element value="my-custom-plugin"/>-->
+			</property>
+		</properties>
+	</rule>
```

Comment:
I wonder if it may cause less confusion / noise, to remove this from this repo-level config, and document how to add a PHPCS config file to the theme directory that extends from this file, and adds in the theme prefix / textdomain config (and likewise for custom plugins). Thoughts @rebeccahum?

Reply:
@GaryJones I feel like if it is already present, customers may be more inclined to use it rather than sifting through documentation.

Reply:
Right, but with this current configuration, it would allow a theme slug is used as textdomain / prefix in a custom plugin, and a custom plugin's textdomain / prefix used in a theme. Ideally, we only want each slug to be in it's relevant entity.

By moving (or rather, giving directions on how to add) the slug/textdomain definition to the relevant theme/foobar or plugins/foobar directory, we keep the scope of that definition to just that directory.

Reply:
> it would allow a theme slug is used as textdomain / prefix in a custom plugin, and a custom plugin's textdomain / prefix used in a theme
Is it possible that a global textdomain / prefix may be what some may want? We could leave a comment with a link to our documentation on an alternative approach on scoping textdomain / prefix?

Reply:
A global textdomain / prefix would be a bit of an anti-pattern IMO. It discourages plugins from being shared between applications, and encourages the use of singular language files to cover the whole application.

I think remove the code that's here, and leave a link to the documentation.

Reply:
OK, we can do that. I don't feel strongly enough to leave it in here.